### DISCUSSION TITLE
pyscenic grn run time for large datasets

### DISCUSSION THREAD

**Discussion by twu-bcm** [aertslab/SCENIC#353] [Python]
Has anyone tried running grn on a larger dataset, say > 200k cells, > 15k genes, and was able to complete the run? How many threads did you use and how much memory/time was required?

I'm trying to estimate resources needed for running pyscenic on a large dataset several times. For a 250k cell chunk, running on 90 threads is taking > 24 hours (still running), which also takes about 400Gb RAM. I think memory limit/worker is 4Gb. Could this be some sort of memory error? Hope to hear your thoughts on this. Thanks!

**Comment by GirayEryilmaz** [comment]
~600k cells, 30 cores, 250GiB mem. took ~ 11 hours.

**Reply by nanakilee** [reply]
LIAR.
